firmware/main.c

```diff
@@ -834,6 +834,148 @@ static void cmd_motor_mode(BaseSequentialStream *chp, int argc, char *argv[]){
 
 }
 
+/* Function for the Linear unit */
+
+static void cmd_motor_lu(BaseSequentialStream *chp, int argc, char *argv[]){
+        uint8_t i,pin2use,direction,start,gear_reduction=48,pOpt1, pOpt2; // Initialize the pins to be used //RL-D-50 has a gear reduction of 1:48
+        uint16_t num_pulses,u; // Unsigned short (0-65535)
+        char *dOpt = NULL, *pOpt = NULL, *mOpt = NULL; // *sOpt = NULL;
+        bool out1 = false, out2 = false; // Variables to store the state of the outputs
+
+        // Parsing
+        for(i = 0; i < argc; i++) {
```

Comment:
This looks like it should use getopt(3).

Reply:
there is no getopt in chibios, and i'd like to not use newlib. firmware already is rather large (50k bin) also the original codebase from chibios does the same, so this is just style mimicri.